spidermon/contrib/scrapy/runners.py

```diff
@@ -73,6 +73,11 @@ def add_action_error(self, action, error):
         super().add_action_error(action, error)
         self.write_item_result(action)
 
+    def declare_settings(self):
+        for key, value in self.spider.crawler.settings.items():
+            if "SPIDERMON" in key:
+                self.log_info(f"{key}: {value}")
```

Comment:
We should consider logging it as debug instead of info.

Reply:
I think info fits the use case better - this data is useful even if the application is running fine. The underlying issue is the myriad places it can be set makes it hard for a user to get this data at all.

Reply:
> If I have multiples suites, why would I want to have the settings printed several times?

It was a limitation with the reporting logic spidermon uses - to handle it in the same way other spidermon reports are made with minimal impact to the code base it ended up triggering per test suite.

For the rest of the feedback you raise an interesting point. More succinctly put - the lack of clarity of settings at run time is a general scrapy problem, not a problem specific to spidermon. Scrapy users in general have this issue regardless of if they are spidermon users. 

With this in mind - do you feel this issue should be closed? Either a seperate extention could be made (which also fixes the redundant reporting + outputs it in a json format) or an option for scrapy to confirm all settings could be more appropriate ways to handle the root cause.

Reply:
I've gone ahead and created a seperate scrapy extention for general logging of settings that covers this use case. This PR and issue can be closed if we don't want a spidermon specific implemention.